### PR TITLE
Expose both head and base SHA for pull requests.

### PR DESCRIPTION
**Pull request by spxtr**
Mostly plumbing. I'm exposing these environment variables to the Jenkins job, `PULL_NUMBER`, `PULL_BASE_REF`, `PULL_PULL_SHA`, and `PULL_BASE_SHA`, and I'm exposing them to users/consumers of http://prow.k8s.io/data.js as `branch`, `head_sha`, and `base_sha` in each entry. I'm not showing it on the actual dash anywhere.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/907)

### CODE REVIEW

**Comment by spxtr** [comment]
BTW I'm not making the associated `bootstrap.py` changes in this PR.

File: prow/cmd/deck/jobs.go [hunk 1]
```diff
@@ -34,6 +34,9 @@ const (
 
 type Job struct {
 	Repo        string `json:"repo"`
+	Branch      string `json:"branch"`
+	HeadSHA     string `json:"head_sha"`
+	BaseSHA     string `json:"base_sha"`
```
Comment (rmmh) [inline prow/cmd/deck/jobs.go]: 'HeadSHA' and 'BaseSHA' are confusing (HEAD has a specific meaning in Git vocabulary). Can you call it "BranchSHA"/"BranchRef" and "BaseRef" instead? (And comment what they mean)

File: prow/cmd/deck/jobs.go [hunk 2]
```diff
@@ -34,6 +34,9 @@ const (
 
 type Job struct {
 	Repo        string `json:"repo"`
+	Branch      string `json:"branch"`
```
Comment (rmmh) [inline prow/cmd/deck/jobs.go]: For batched merges, I'll need to expand this to being a map of branch names (pull/1234) to SHAs. This is fine for now.

File: prow/cmd/deck/jobs.go (hunk 1 above)
Comment (spxtr) [inline prow/cmd/deck/jobs.go]: Sure. I just went with what is in the webhook payload, but I agree it's not clear.

**Comment by spxtr** [comment]
Renamed the things. I'd like to do multiple PRs in a future PR.

**Comment by rmmh** [comment]
LGTM

**Pull request merged**